src/libraries/Microsoft.Extensions.Primitives/src/CompositeChangeToken.cs

```diff
@@ -102,10 +108,10 @@ private void EnsureCallbacksInitialized()
             }
         }
 
-        private static void OnChange(object state)
+        private static void OnChange(object? state)
         {
-            var compositeChangeTokenState = (CompositeChangeToken)state;
-            if (compositeChangeTokenState._cancellationTokenSource == null)
+            var compositeChangeTokenState = (CompositeChangeToken?)state;
+            if (compositeChangeTokenState?._cancellationTokenSource == null)
```

Comment:
This is actually changing behavior since there is a new `null` check getting inserted here. I think it would be better to `Debug.Assert` that `state != null` since the only callers of this method pass `this` for `state`.